File: tools/parity/dispatch/terminal-tab-id.ts
// TS dispatch for the terminal-tab-id parity module: maps the shared vector
// function names to the real `src/shared/terminal-tab-id.ts` exports so the
// harness compares the live TS reference against the Rust port.

import {
  isValidHostTerminalTabId,
  isValidTerminalTabId
} from '../../../src/shared/terminal-tab-id'

export function dispatch(fn: string, input: unknown): unknown {
  switch (fn) {
    case 'isValidTerminalTabId':
      return isValidTerminalTabId(input as string)
    case 'isValidHostTerminalTabId':
      return isValidHostTerminalTabId(input as string)
    default:
      throw new Error(`unknown function ${fn}`)
  }
}

File: rust/crates/orca-dispatch/src/modules/terminal_tab_id.rs
//! Parity dispatch for `orca_core::terminal_tab_id` vs
//! `src/shared/terminal-tab-id.ts`.

use orca_core::terminal_tab_id::{is_valid_host_terminal_tab_id, is_valid_terminal_tab_id};
use serde_json::{json, Value};

pub fn dispatch(function: &str, input: &Value) -> Value {
    // Both functions take a single string `value`; vectors carry it as the raw
    // JSON string. A non-string is a vector bug, not a port divergence.
    let value = input.as_str().unwrap_or_default();
    match function {
        "isValidTerminalTabId" => Value::Bool(is_valid_terminal_tab_id(value)),
        "isValidHostTerminalTabId" => Value::Bool(is_valid_host_terminal_tab_id(value)),
        other => json!({ "__parity_error__": format!("unknown function {other}") }),
    }
}
